### DISCUSSION TITLE
question about future

### DISCUSSION THREAD

**Discussion by danielbraun89** [openapi-json-schema-tools/openapi-json-schema-generator#36] [General]
I was trying to decide which python generator to use the other day (just for a personal project) I'm not some important company so you can take my opinion as it is, also english not first language 

it seems the python generator of oneAPITools was mostly written by @spacether , and because he went to open this repo I assume this repo will contain the most up to date python generator in the future, not sure who will continue the work on python in the original repo but looking at the git activity spacether will be hard to replace... also I like the full jsonschema compatability so I guess this generator is what I want to use,

however wouldnt it be smarter to have this project as some kind of plugin auxiliary generator to the main oneAPITools generator,? (dont know if such a thing exists)    
this will make it easier to maintain this repo, as currently it will have to keep merging core changes from the original repo (CLI etc)  and there is no garaunteee other langueges will switch, considering original repo still enjoy the mindshare and sponsors and funding.  

what Im trying to say is I dont want to see this project die out because of non-technical reasons because I do believe technically its in the best path,  it seems to have taken some "fighting windmills" approach which rarely succeed , is there something here im missing

**Comment by spacether** [comment]
Because this project and the other rely on the same core code some kind of plug in would be possible if the same core code were used. It would be complicated to maintain that though.
Each generator works similar to a plugin in that repo and this one. Generators consolidate spec data using public interface methods and render their data into generated code using generator specific mustache or handlebars templates.
If you prefer to have this generator stay in openapi-generator, that is great feedback to leave on the [6.2.0 release PR](https://github.com/OpenAPITools/openapi-generator/pull/13501). You can read about the reasons for creation of this repo [here](https://github.com/openapi-json-schema-tools/openapi-json-schema-generator#51---history-of-openapi-json-schema-generator).

> it (this repo) will have to keep merging core changes from the original repo (CLI etc)

No it will not. They will be separate projects. If people want to add future new openapi-generator features here they can but it is not a requirement.

Can you elaborate on your ` it seems to have taken some "fighting windmills" approach`? What are the windmills that the python generator code sees or is fighting?

**Reply by spacether** [reply]
If the `fighting windmills` was referring to the _oapg suffixes, they will be removed and replaced with a trailing `_` to prevent name collisions in the v2 release per: https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/pull/112
